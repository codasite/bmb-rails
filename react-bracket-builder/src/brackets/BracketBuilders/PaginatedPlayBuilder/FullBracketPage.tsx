import React, { useContext } from 'react'
import darkBracketBg from '../../shared/assets/bracket-bg-dark.png'
import lightBracketBg from '../../shared/assets/bracket-bg-light.png'
import { ActionButton } from '../../shared/components/ActionButtons'
import { PickableBracket } from '../../shared/components/Bracket'
import { ThemeSelector } from '../../shared/components'
import { ScaledBracket } from '../../shared/components/Bracket/ScaledBracket'
import { ReactComponent as EditIcon } from '../../shared/assets/edit-icon.svg'
import { PlayBuilderButtons } from '../PlayBracketBuilder/PlayBuilderButtons'
import { PlayBuilderProps } from '../PlayBracketBuilder/types'
import { DarkModeContext } from '../../shared/context/context'
import { BracketHeaderTag } from '../BracketHeaderTag'

interface FullBracketPageProps extends PlayBuilderProps {
  onEditClick?: () => void
}

export const FullBracketPage = (props: FullBracketPageProps) => {
  const { onEditClick, matchTree } = props
  const { darkMode } = useContext(DarkModeContext)

  const canEdit = !!onEditClick
  const processing = props.processingAddToApparel || props.processingSubmitPicks
  return (
    <div
      className={`wpbb-reset tw-min-h-screen tw-flex tw-flex-col tw-justify-center tw-items-center tw-uppercase tw-bg-no-repeat tw-bg-top tw-bg-cover ${
        darkMode ? ' tw-dark' : ''
      }`}
      style={{
        backgroundImage: `url(${darkMode ? darkBracketBg : lightBracketBg})`,
      }}
    >
      <div className="tw-flex tw-flex-col tw-justify-between tw-items-center tw-mx-auto tw-flex-grow tw-mt-60 tw-mb-80">
        <div className="tw-self-center">
          <div className="tw-mb-10">
            <BracketHeaderTag
              text={`Voting Round ${matchTree.liveRoundIndex + 1}`}
              color="green"
            />
          </div>
          <ThemeSelector />
        </div>

        {matchTree && (
          <ScaledBracket
            BracketComponent={PickableBracket}
            matchTree={matchTree}
          />
        )}
        <div className="tw-flex tw-self-stretch tw-flex-col tw-gap-10 tw-items-stretch tw-px-4">
          {canEdit && (
            <ActionButton
              variant="white"
              onClick={onEditClick}
              disabled={processing}
              borderWidth={0}
            >
              <EditIcon />
              <span>Edit</span>
            </ActionButton>
          )}
          <PlayBuilderButtons {...props} />
        </div>
      </div>
    </div>
  )
}
