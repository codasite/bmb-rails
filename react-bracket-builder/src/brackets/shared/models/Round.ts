import { Nullable } from '../../../utils/types'
import { MatchNode } from './operations/MatchNode'
import { MatchRepr } from '../api/types/bracket'
import { Team } from './Team'

export class Round {
  index: number
  depth: number
  matches: Array<Nullable<MatchNode>>
  constructor(
    index: number,
    depth: number,
    matches: Array<Nullable<MatchNode>> = []
  ) {
    this.index = index
    this.depth = depth
    this.matches = matches
  }
  allPicked(): boolean {
    return this.matches.every((match) => {
      if (match === null) {
        return true
      }
      return match.team1Wins || match.team2Wins
    })
  }
  getTeam(teamId: number): Nullable<Team> {
    for (const [index, match] of this.matches.entries()) {
      if (match === null) {
        continue
      }
      const team1 = match.getTeam1()
      const team2 = match.getTeam2()

      if (team1?.id === teamId) {
        team1.side = index < this.matches.length / 2 ? 'left' : 'right'
        return team1
      }

      if (team2?.id === teamId) {
        team2.side = index < this.matches.length / 2 ? 'left' : 'right'
        return team2
      }
    }

    return null
  }
  serialize(): Nullable<MatchRepr>[] {
    return this.matches.map((match, i) => {
      if (match === null) {
        return null
      }
      return match.serialize()
    })
  }
}
