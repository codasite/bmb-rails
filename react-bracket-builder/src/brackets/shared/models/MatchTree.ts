import { Nullable } from '../../../utils/types'
import { BracketRes, MatchPick, MatchReq, MatchTreeRepr } from '../api'
import { WildcardPlacement } from './WildcardPlacement'
import { Team } from './Team'
import { MatchNode } from './operations/MatchNode'
import { Round } from './Round'
import { matchReprFromNumTeams } from './operations/MatchReprFromNumTeams'
import { getNumRounds } from './operations/GetNumRounds'
import { matchReprFromRes } from './operations/MatchReprFromRes'
import { linkNodes } from './operations/LinkNodes'

export class MatchTree {
  rounds: Round[]
  private numTeams: number
  private wildcardPlacement?: WildcardPlacement
  isVoting: boolean
  liveRoundIndex: number

  constructor(
    rounds: Round[] = [],
    wildcardPlacement?: WildcardPlacement,
    isVoting: boolean = false,
    liveRoundIndex: number = 0
  ) {
    linkNodes(rounds)
    this.rounds = rounds
    this.wildcardPlacement = wildcardPlacement
    this.isVoting = isVoting
    this.liveRoundIndex = liveRoundIndex
  }

  getRootMatch(): Nullable<MatchNode> {
    const lastRound = this.rounds[this.rounds.length - 1]
    if (!lastRound) {
      return null
    }
    return lastRound.matches[0]
  }

  clone(): MatchTree {
    return MatchTree.deserialize(this.serialize())
  }

  equals(other: MatchTree): boolean {
    return (
      JSON.stringify(this.serialize()) === JSON.stringify(other.serialize())
    )
  }

  /**
   * Returns the total number of POSSIBLE teams in the tournament.
   * This should reflect the number of teams that would be used to create the same structure with MatchTree.fromNumTeams()
   */
  getNumTeams(): number {
    if (this.numTeams) {
      return this.numTeams
    }
    // count all non null matches in the tree
    const numMatches = this.rounds.reduce((acc, round) => {
      return (
        acc +
        round.matches.reduce((acc, match) => {
          if (match) {
            return acc + 1
          }
          return acc
        }, 0)
      )
    }, 0)

    const numTeams = numMatches + 1
    this.numTeams = numTeams

    return numTeams
  }

  getWildcardPlacement(): WildcardPlacement | undefined {
    return this.wildcardPlacement
  }

  serialize(): MatchTreeRepr {
    const tree = this
    const rounds = tree.rounds.map((round) => {
      return round.serialize()
    })
    return {
      rounds: rounds,
      wildcardPlacement: tree.wildcardPlacement,
      isVoting: tree.isVoting,
      liveRoundIndex: tree.liveRoundIndex,
    }
  }

  advanceTeam = (
    roundIndex: number,
    matchIndex: number,
    left: boolean,
    requireComplete: boolean = false
  ) => {
    if (requireComplete && roundIndex > 0) {
      const prevRound = this.rounds[roundIndex - 1]
      if (prevRound && !prevRound.allPicked()) {
        return
      }
    }
    const round = this.rounds[roundIndex]
    const match = round.matches[matchIndex]
    if (!match) {
      return
    }
    match.team1Wins = false
    match.team2Wins = false
    if (left) {
      match.team1Wins = true
    } else {
      match.team2Wins = true
    }
  }

  anyPicked = (): boolean => {
    return this.rounds.some((round) => {
      return round.matches.some((match) => {
        if (!match) {
          return false
        }
        return match.team1Wins || match.team2Wins
      })
    })
  }

  allPicked = (): boolean => {
    if (this.isVoting) {
      return this.rounds[this.liveRoundIndex].allPicked()
    }
    return this.rounds.every((round) => {
      return round.allPicked()
    })
  }

  allTeamsAdded = (): boolean => {
    return this.everyMatch((match) => {
      let hasNeededTeams = true
      if (!match.left && !match.getTeam1()?.name) {
        hasNeededTeams = false
      }
      if (hasNeededTeams && !match.right && !match.getTeam2()?.name) {
        hasNeededTeams = false
      }
      return hasNeededTeams
    })
  }

  toMatchReq = (): MatchReq[] => {
    const matches: MatchReq[] = []
    this.forEachMatch((match, matchIndex, roundIndex) => {
      const matchReq: MatchReq = {
        roundIndex: match.roundIndex,
        matchIndex: match.matchIndex,
      }
      if (!match.left) {
        matchReq.team1 = match.getTeam1()?.toTeamReq()
      }
      if (!match.right) {
        matchReq.team2 = match.getTeam2()?.toTeamReq()
      }
      if (matchReq.team1 || matchReq.team2) {
        matches.push(matchReq)
      }
    })
    return matches
  }

  toMatchPicks = (): MatchPick[] => {
    const picks: MatchPick[] = []
    this.rounds.forEach((round, roundIndex) => {
      round.matches.forEach((match, matchIndex) => {
        if (!match) {
          return
        }
        const { team1Wins, team2Wins } = match
        const team1 = match.getTeam1()
        const team2 = match.getTeam2()
        const team1Winner = team1Wins && team1
        const team2Winner = team2Wins && team2

        if (team1Winner && team1.id) {
          picks.push({
            roundIndex,
            matchIndex,
            winningTeamId: team1.id,
          })
        } else if (team2Winner && team2.id) {
          picks.push({
            roundIndex,
            matchIndex,
            winningTeamId: team2.id,
          })
        }
      })
    })
    return picks
  }

  forEachMatch = (
    callback: (match: MatchNode, matchIndex: number, roundIndex: number) => void
  ) => {
    this.rounds.forEach((round, roundIndex) => {
      round.matches.forEach((match, matchIndex) => {
        if (!match) {
          return
        }
        callback(match, matchIndex, roundIndex)
      })
    })
  }

  everyMatch = (
    callback: (
      match: MatchNode,
      matchIndex: number,
      roundIndex: number
    ) => boolean
  ) => {
    return this.rounds.every((round, roundIndex) => {
      return round.matches.every((match, matchIndex) => {
        if (!match) {
          return true
        }
        return callback(match, matchIndex, roundIndex)
      })
    })
  }

  findMatch = (
    callback: (match: Nullable<MatchNode>) => boolean
  ): Nullable<MatchNode> => {
    let foundMatch: Nullable<MatchNode> = null
    this.rounds.some((round) => {
      return round.matches.some((match) => {
        if (callback(match)) {
          foundMatch = match
          return true
        }
        return false
      })
    })
    return foundMatch
  }

  static fromNumTeams(
    numTeams: number,
    wildcardPlacement: WildcardPlacement = WildcardPlacement.Top
  ): MatchTree {
    const matches = matchReprFromNumTeams(numTeams, wildcardPlacement)
    return MatchTree.deserialize({
      rounds: matches,
      wildcardPlacement,
    })
  }

  static fromMatchRes(
    bracket: Pick<
      BracketRes,
      'matches' | 'numTeams' | 'isVoting' | 'liveRoundIndex'
    >,
    wildcardPlacement?: WildcardPlacement
  ): MatchTree | null {
    const numRounds = getNumRounds(bracket.numTeams)

    const nestedMatches = matchReprFromRes(numRounds, bracket.matches)
    return MatchTree.deserialize({
      rounds: nestedMatches,
      wildcardPlacement,
      isVoting: bracket.isVoting,
      liveRoundIndex: bracket.liveRoundIndex,
    })
  }

  static fromPicks(
    bracket: Pick<
      BracketRes,
      'matches' | 'numTeams' | 'isVoting' | 'liveRoundIndex'
    >,
    picks: MatchPick[],
    wildcardPlacement?: WildcardPlacement
  ): MatchTree | null {
    const matchTree = MatchTree.fromMatchRes(bracket, wildcardPlacement)
    if (!matchTree) {
      return null
    }
    for (const pick of picks) {
      const { roundIndex, matchIndex, winningTeamId } = pick
      const match = matchTree.rounds[roundIndex].matches[matchIndex]
      if (!match) {
        throw new Error('Match not found')
      }
      match._pick = pick

      const team1 = match.getTeam1()
      const team2 = match.getTeam2()
      if (team1?.id === winningTeamId) {
        match.team1Wins = true
      } else if (team2?.id === winningTeamId) {
        match.team2Wins = true
      } else {
        throw new Error('Match does not have winningTeam from pick')
      }
    }
    return matchTree
  }

  static deserialize(matchTreeRepr: MatchTreeRepr): MatchTree {
    const { rounds: matchRes, wildcardPlacement } = matchTreeRepr
    const rounds = matchRes.map((round, roundIndex) => {
      const depth = matchRes.length - roundIndex - 1
      const newRound = new Round(roundIndex, depth)
      const matches = round.map((match, matchIndex) => {
        if (match === null) {
          return null
        }
        const {
          id,
          team1: team1Repr,
          team2: team2Repr,
          team1Wins,
          team2Wins,
          pick,
        } = match
        const team1 = team1Repr ? new Team(team1Repr.name, team1Repr.id) : null
        const team2 = team2Repr ? new Team(team2Repr.name, team2Repr.id) : null
        // const newMatch = new MatchNode(roundIndex, matchIndex, depth, match.id, team1, team2, team1Wins, team2Wins)
        const newMatch = new MatchNode({
          id,
          matchIndex,
          roundIndex,
          team1,
          team2,
          team1Wins,
          team2Wins,
          left: null,
          right: null,
          parent: null,
          depth,
          pick,
        })
        return newMatch
      })
      newRound.matches = matches
      return newRound
    })
    const tree = new MatchTree(
      rounds,
      wildcardPlacement,
      matchTreeRepr.isVoting,
      matchTreeRepr.liveRoundIndex
    )
    return tree
  }
}
