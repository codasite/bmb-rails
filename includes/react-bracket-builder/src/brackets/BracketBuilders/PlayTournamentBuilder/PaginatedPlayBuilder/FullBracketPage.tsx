import React, { useState, useContext } from 'react';
import darkBracketBg from '../../../shared/assets/bracket-bg-dark.png'
import lightBracketBg from '../../../shared/assets/bracket-bg-light.png'
import { MatchTree } from '../../../shared/models/MatchTree';
import { ActionButton } from '../../../shared/components/ActionButtons';
import { PickableBracket } from '../../../shared/components/Bracket';
import { DarkModeContext } from '../../../shared/context';
import { ThemeSelector } from '../../../shared/components';

interface FullBracketPageProps {
	onApparelClick: () => void;
	matchTree?: MatchTree;
	darkMode?: boolean;
	setDarkMode?: (darkMode: boolean) => void;
}

export const FullBracketPage = (props: FullBracketPageProps) => {
	const {
		onApparelClick,
		matchTree,
		darkMode,
		setDarkMode,
	} = props;


	return (
		<div className={`wpbb-reset tw-flex tw-uppercase tw-bg-no-repeat tw-bg-top tw-bg-cover${darkMode ? ' tw-dark' : ''}`} style={{ 'backgroundImage': `url(${darkMode ? darkBracketBg : lightBracketBg})` }}>
			<div className='tw-flex tw-flex-col tw-justify-center px-60 tw-max-w-[268px] tw-m-auto'>
				<ThemeSelector darkMode={darkMode} setDarkMode={setDarkMode} />
				{matchTree &&
					<div className='tw-flex tw-flex-col tw-justify-center tw-items-center tw-pb-60 tw-h-[350px]'>
						<div className='tw-scale-30'>
							<PickableBracket
								matchTree={matchTree}
								lineStyle={{
									className: `!tw-border-t-white !tw-border-t-[.4px]`,
								}}
							/>
						</div>
					</div>
				}
				<ActionButton
					height={48}
					borderColor='green'
					borderWidth={4}
					backgroundColor='green/15'
					fontSize={24}
					fontWeight={700}
					onClick={onApparelClick}
					textColor='white'
					borderRadius={8}
				>Start</ActionButton>
				{/* <div className={'wpbb-slogan-container'}>
				<span className={'wpbb-slogan-text'}>WHO YOU GOT?</span>
			</div>
			<div className='wpbb-bracket-image-container'>
				<img src='https://wpbb-bracket-images.s3.amazonaws.com/bracket-m7g1t-dark-center-cropped.png'></img>
			</div>
			<ActionButton label='START' onClick={onStart} /> */}
			</div>
		</div>
	)
}