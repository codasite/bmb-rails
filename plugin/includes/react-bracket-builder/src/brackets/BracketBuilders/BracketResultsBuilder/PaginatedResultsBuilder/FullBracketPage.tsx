import React, { useState, useContext } from 'react'
import darkBracketBg from '../../../shared/assets/bracket-bg-dark.png'
import lightBracketBg from '../../../shared/assets/bracket-bg-light.png'
import { MatchTree } from '../../../shared/models/MatchTree'
import { ActionButton } from '../../../shared/components/ActionButtons'
import { ResultsBracket } from '../../../shared/components/Bracket'
import { DarkModeContext } from '../../../shared/context'
import { ThemeSelector } from '../../../shared/components'
import { ScaledBracket } from '../../../shared/components/Bracket/ScaledBracket'
import { bracketApi } from '../../../shared/api/bracketApi'


interface FullBracketPageProps {
  onApparelClick: () => void
  matchTree?: MatchTree
  darkMode?: boolean
  setDarkMode?: (darkMode: boolean) => void
  processing?: boolean
  myBracketsUrl?: string
}

export const FullBracketPage = (props: FullBracketPageProps) => {
  const { myBracketsUrl, matchTree, darkMode, setDarkMode, processing } = props
  const [notifyParticipants, setNotifyParticipants] = useState(true)
  const [bracketId, setBracketId] = useState(0)

  console.log('darkMode', darkMode)

  const handleUpdatePicks = () => {
    if (matchTree) {
      const picks = matchTree.toMatchPicks()
      if (!picks || picks.length === 0) return
      const complete = matchTree.allPicked()
      const data = {
        results: picks,
        updateNotifyPlayers: notifyParticipants,
      }
      bracketApi
        .updateBracket(bracketId, data)
        .then((res) => {
          console.log(res)
        })
        .catch((err) => {
          console.log(err)
        })
        .finally(() => {
          if (myBracketsUrl) window.location.href = myBracketsUrl || ''
        })
    }
  }
  

  return (
    <div
      className={`wpbb-reset tw-min-h-screen tw-flex tw-flex-col tw-justify-center tw-items-center tw-uppercase tw-bg-no-repeat tw-bg-top tw-bg-cover${
        darkMode ? ' tw-dark' : ''
      }`}
      style={{
        backgroundImage: `url(${darkMode ? darkBracketBg : lightBracketBg})`,
      }}
    >
      <div className="tw-flex tw-flex-col tw-justify-between tw-max-w-[268px] tw-max-h-[500px] tw-mx-auto tw-flex-grow tw-my-60">
        <ThemeSelector darkMode={darkMode} setDarkMode={setDarkMode} />
        {matchTree && (
          <ScaledBracket
            BracketComponent={ResultsBracket}
            matchTree={matchTree}
          />
        )}
        <ActionButton
          variant="small-yellow"
          darkMode={darkMode}
          onClick={handleUpdatePicks}
          disabled={processing || !matchTree?.allPicked()}
          fontSize={16}
        >
          complete tournament
        </ActionButton>
      </div>
    </div>
  )
}
