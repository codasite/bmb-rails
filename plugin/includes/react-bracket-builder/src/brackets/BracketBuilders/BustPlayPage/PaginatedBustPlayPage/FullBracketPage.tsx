import React, { useState, useContext } from 'react'
import darkBracketBg from '../../../shared/assets/bracket-bg-dark.png'
import lightBracketBg from '../../../shared/assets/bracket-bg-light.png'
import { MatchTree } from '../../../shared/models/MatchTree'
import { ActionButton } from '../../../shared/components/ActionButtons'
import { ResultsBracket } from '../../../shared/components/Bracket'
import { DarkModeContext } from '../../../shared/context'
import { ThemeSelector } from '../../../shared/components'
import { ScaledBracket } from '../../../shared/components/Bracket/ScaledBracket'
import { bracketApi } from '../../../shared/api/bracketApi'


interface FullBracketPageProps {
  matchTree?: MatchTree
  darkMode?: boolean
  setDarkMode?: (darkMode: boolean) => void
  processing?: boolean
  myBracketsUrl?: string
  handleUpdatePicks?: () => void
}

export const FullBracketPage = (props: FullBracketPageProps) => {
  const { myBracketsUrl, matchTree, darkMode, setDarkMode, processing, handleUpdatePicks } = props
  const [notifyParticipants, setNotifyParticipants] = useState(true)
  const [bracketId, setBracketId] = useState(0)

  console.log('darkMode', darkMode)

  return (
    <div
      className={`wpbb-reset tw-min-h-screen tw-flex tw-flex-col tw-justify-center tw-items-center tw-uppercase tw-bg-no-repeat tw-bg-top tw-bg-cover${
        darkMode ? ' tw-dark' : ''
      }`}
      style={{
        backgroundImage: `url(${darkMode ? darkBracketBg : lightBracketBg})`,
      }}
    >
      <div className="tw-flex tw-flex-col tw-justify-between tw-max-w-[268px] tw-max-h-[500px] tw-mx-auto tw-flex-grow tw-my-60">
        <ThemeSelector darkMode={darkMode} setDarkMode={setDarkMode} />
        {matchTree && (
          <ScaledBracket
            BracketComponent={ResultsBracket}
            matchTree={matchTree}
          />
        )}
        <ActionButton
          variant="small-yellow"
          darkMode={darkMode}
          onClick={handleUpdatePicks}
          disabled={processing || !matchTree?.allPicked()}
          fontSize={16}
          backgroundColor='yellow'
          textColor='dd-blue'
        >
          {matchTree.allPicked() ? 'Complete Bracket' : 'Update Picks'}
        </ActionButton>
      </div>
    </div>
  )
}
