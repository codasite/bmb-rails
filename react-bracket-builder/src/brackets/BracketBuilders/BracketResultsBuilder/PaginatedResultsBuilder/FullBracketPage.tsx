import React, { useContext } from 'react'
import darkBracketBg from '../../../shared/assets/bracket-bg-dark.png'
import lightBracketBg from '../../../shared/assets/bracket-bg-light.png'
import { MatchTree } from '../../../shared/models/MatchTree'
import { ActionButton } from '../../../shared/components/ActionButtons'
import { ResultsBracket } from '../../../shared/components/Bracket'
import { ScaledBracket } from '../../../shared/components/Bracket/ScaledBracket'
import { ReactComponent as EditIcon } from '../../../shared/assets/edit-icon.svg'
import { Checkbox } from '../Checkbox'
import { BracketResultsBuilderContext } from '../context'
import { DarkModeContext } from '../../../shared/context/context'

interface FullBracketPageProps {
  matchTree?: MatchTree
  processing?: boolean
  handleUpdatePicks: () => void
  onEditClick: () => void
}

export const FullBracketPage = (props: FullBracketPageProps) => {
  const { matchTree, processing, onEditClick, handleUpdatePicks } = props

  const { notifyParticipants, toggleNotifyParticipants } = useContext(
    BracketResultsBuilderContext
  )
  const { darkMode } = useContext(DarkModeContext)

  return (
    <div
      className={`wpbb-reset tw-min-h-screen tw-flex tw-flex-col tw-justify-center tw-items-center tw-uppercase tw-bg-no-repeat tw-bg-top tw-bg-cover ${
        darkMode ? ' tw-dark' : ''
      }`}
      style={{
        backgroundImage: `url(${darkMode ? darkBracketBg : lightBracketBg})`,
      }}
    >
      <div className="tw-flex tw-flex-col tw-justify-between tw-max-w-[268px] tw-max-h-[500px] tw-mx-auto tw-flex-grow tw-my-60">
        {matchTree && (
          <ScaledBracket
            BracketComponent={ResultsBracket}
            matchTree={matchTree}
          />
        )}
        <div className="tw-flex tw-flex-col tw-gap-10">
          <ActionButton
            variant="white"
            onClick={onEditClick}
            disabled={processing}
            borderWidth={1}
          >
            <EditIcon />
            <span>Edit</span>
          </ActionButton>
          <ActionButton
            variant="yellow"
            size="small"
            onClick={handleUpdatePicks}
            disabled={processing}
            fontSize={16}
          >
            {matchTree.allPicked() ? 'Complete Bracket' : 'Update Picks'}
          </ActionButton>
        </div>
        <div className="tw-flex tw-items-center tw-justify-center tw-gap-[16px] tw-mt-[52px]">
          <Checkbox
            id="notify-participants-check"
            checked={notifyParticipants}
            onChange={toggleNotifyParticipants}
            height={24}
            width={24}
          />
          <label
            htmlFor="notify-participants-check"
            className="tw-font-500 tw-text-16 tw-items-center"
          >
            Notify Participants
          </label>
        </div>
      </div>
    </div>
  )
}
