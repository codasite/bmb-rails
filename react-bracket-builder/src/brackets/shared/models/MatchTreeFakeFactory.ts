import { MatchTree } from './MatchTree'

export function fourTeamMatchTree() {
  const matches = [
    [
      {
        id: 9,
        roundIndex: 0,
        matchIndex: 0,
        team1: { id: 17, name: 'Team 1' },
        team2: { id: 18, name: 'Team 2' },
        team1Wins: true,
      },
      {
        id: 10,
        roundIndex: 0,
        matchIndex: 1,
        team1: { id: 19, name: 'Team 3' },
        team2: { id: 20, name: 'Team 4' },
        team1Wins: true,
      },
    ],
    [{ roundIndex: 1, matchIndex: 1, team2Wins: true }],
  ]
  return MatchTree.deserialize({ rounds: matches })
}
