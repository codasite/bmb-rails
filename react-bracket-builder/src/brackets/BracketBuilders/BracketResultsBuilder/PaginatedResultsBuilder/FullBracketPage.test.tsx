import React from 'react'
import { render } from '@testing-library/react'
import { FullBracketPage } from './FullBracketPage'
import { fourTeamMatchTree } from '../../../shared/models/MatchTreeFakeFactory'

describe('FullBracketPage', () => {
  test('renders page correctly', () => {
    const noop = () => {}
    const { asFragment } = render(
      <FullBracketPage
        matchTree={fourTeamMatchTree()}
        handleUpdatePicks={noop}
        onEditClick={noop}
      />
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
