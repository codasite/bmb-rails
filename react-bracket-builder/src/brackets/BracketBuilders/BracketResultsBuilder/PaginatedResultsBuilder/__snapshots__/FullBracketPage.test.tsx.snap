// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FullBracketPage renders page correctly 1`] = `
<DocumentFragment>
  <div
    class="wpbb-reset tw-min-h-screen tw-flex tw-flex-col tw-justify-center tw-items-center tw-uppercase tw-bg-no-repeat tw-bg-top tw-bg-cover  tw-dark"
    style="background-image: url(undefined);"
  >
    <div
      class="tw-flex tw-flex-col tw-justify-between tw-max-w-[268px] tw-max-h-[500px] tw-mx-auto tw-flex-grow tw-my-60"
    >
      <div
        class="tw-flex tw-flex-col tw-justify-center tw-items-center"
        style="height: 0px; width: 210px;"
      >
        <div
          style="transform: scale(0.3);"
        >
          <div
            class="tw-flex tw-flex-col tw-dark wpbb-default-bracket tw-relative"
          >
            <div
              class="tw-flex tw-flex-col tw-justify-end"
              style="margin-bottom: 95px; min-height: 0;"
            >
              <div
                class="tw-flex tw-flex-col tw-items-center"
                style="gap: 24px;"
              >
                <div
                  class="wpbb-team-1-0-winner tw-relative tw-flex tw-justify-center tw-items-center tw-whitespace-nowrap tw-leading-none tw-uppercase tw-font-sans tw-w-[297px] tw-h-[52px] tw-text-dd-blue tw-font-700 tw-bg-green"
                  style="border-width: 0px;"
                >
                  <span
                    style="font-size: 32px; transform: scale(1); visibility: hidden;"
                  >
                    Team 3
                  </span>
                </div>
              </div>
            </div>
            <div
              class="tw-flex tw-flex-col tw-justify-center tw-h-100"
            >
              <div
                class="tw-flex tw-justify-between"
                style="width: 700px;"
              >
                <div
                  class="wpbb-default-match-column tw-flex tw-flex-col tw-justify-center tw-gap-[80px]"
                >
                  <div
                    class="tw-flex tw-flex-col tw-gap-[20px] tw-translate-y-[0px]"
                  >
                    <button
                      class="wpbb-team-0-0-left tw-relative tw-flex tw-justify-center tw-items-center tw-whitespace-nowrap tw-leading-none tw-uppercase tw-font-sans tw-w-[115px] tw-h-[28px] tw-text-dd-blue tw-font-500 tw-bg-green tw-cursor-pointer"
                      style="border-width: 0px;"
                    >
                      <span
                        style="font-size: 16px; transform: scale(1); visibility: hidden;"
                      >
                        Team 1
                      </span>
                    </button>
                    <button
                      class="wpbb-team-0-0-right tw-relative tw-flex tw-justify-center tw-items-center tw-whitespace-nowrap tw-leading-none tw-uppercase tw-font-sans tw-w-[115px] tw-h-[28px] tw-text-white tw-font-500 tw-bg-transparent tw-border-solid tw-border-white/50 tw-cursor-pointer"
                      style="border-width: 2px;"
                    >
                      <span
                        style="font-size: 16px; transform: scale(1); visibility: hidden;"
                      >
                        Team 2
                      </span>
                    </button>
                  </div>
                </div>
                <div
                  class="wpbb-default-match-column tw-flex tw-flex-col tw-justify-center tw-gap-[236px]"
                >
                  <div
                    class="tw-flex tw-flex-col tw-gap-[20px] tw-translate-y-[-48px]"
                  >
                    <button
                      class="wpbb-team-1-0-left tw-relative tw-flex tw-justify-center tw-items-center tw-whitespace-nowrap tw-leading-none tw-uppercase tw-font-sans tw-w-[115px] tw-h-[28px] tw-text-white tw-font-500 tw-bg-transparent tw-border-solid tw-border-white/50 tw-cursor-pointer"
                      style="border-width: 2px;"
                    >
                      <span
                        style="font-size: 16px; transform: scale(1); visibility: hidden;"
                      >
                        Team 1
                      </span>
                    </button>
                    <button
                      class="wpbb-team-1-0-right tw-relative tw-flex tw-justify-center tw-items-center tw-whitespace-nowrap tw-leading-none tw-uppercase tw-font-sans tw-w-[115px] tw-h-[28px] tw-text-dd-blue tw-font-500 tw-bg-green tw-cursor-pointer"
                      style="border-width: 0px;"
                    >
                      <span
                        style="font-size: 16px; transform: scale(1); visibility: hidden;"
                      >
                        Team 3
                      </span>
                    </button>
                  </div>
                </div>
                <div
                  class="wpbb-default-match-column tw-flex tw-flex-col tw-justify-center tw-gap-[80px]"
                >
                  <div
                    class="tw-flex tw-flex-col tw-gap-[20px] tw-translate-y-[0px]"
                  >
                    <button
                      class="wpbb-team-0-1-left tw-relative tw-flex tw-justify-center tw-items-center tw-whitespace-nowrap tw-leading-none tw-uppercase tw-font-sans tw-w-[115px] tw-h-[28px] tw-text-dd-blue tw-font-500 tw-bg-green tw-cursor-pointer"
                      style="border-width: 0px;"
                    >
                      <span
                        style="font-size: 16px; transform: scale(1); visibility: hidden;"
                      >
                        Team 3
                      </span>
                    </button>
                    <button
                      class="wpbb-team-0-1-right tw-relative tw-flex tw-justify-center tw-items-center tw-whitespace-nowrap tw-leading-none tw-uppercase tw-font-sans tw-w-[115px] tw-h-[28px] tw-text-white tw-font-500 tw-bg-transparent tw-border-solid tw-border-white/50 tw-cursor-pointer"
                      style="border-width: 2px;"
                    >
                      <span
                        style="font-size: 16px; transform: scale(1); visibility: hidden;"
                      >
                        Team 4
                      </span>
                    </button>
                  </div>
                </div>
              </div>
            </div>
            <div
              style="margin-top: 20px; min-height: 0;"
            >
              <div
                class="tw-flex tw-flex-col tw-gap-20 tw-justify-between tw-items-center tw-font-700 tw-whitespace-nowrap "
              >
                <span
                  class="tw-text-36 tw-text-dd-blue dark:tw-text-white"
                >
                  Who You Got?
                </span>
                <svg
                  class="tw-w-[124px] tw-text-black/25 dark:tw-text-white"
                />
              </div>
            </div>
            <div
              class="tw-absolute"
            >
              <div
                class="react-lineto-placeholder"
              />
              <div
                class="react-lineto-placeholder"
              />
              <div
                class="react-lineto-placeholder"
              />
              <div
                class="react-lineto-placeholder"
              />
            </div>
            <div
              class="tw-absolute"
            >
              <div
                class="react-lineto-placeholder"
              />
              <div
                class="react-lineto-placeholder"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="tw-flex tw-flex-col tw-gap-10"
      >
        <button
          class="tw-flex tw-flex-row tw-items-center tw-justify-center tw-font-sans tw-uppercase tw-whitespace-nowrap tw-cursor-pointer disabled:tw-cursor-default tw-leading-none tw-cursor-pointer tw-bg-transparent tw-text-white tw-gap-10 tw-text-24 tw-font-700 tw-rounded-8 tw-h-[48px] tw-border-solid tw-border tw-border-white"
        >
          <svg />
          <span>
            Edit
          </span>
        </button>
        <button
          class="tw-flex tw-flex-row tw-items-center tw-justify-center tw-font-sans tw-uppercase tw-whitespace-nowrap tw-cursor-pointer disabled:tw-cursor-default tw-leading-none tw-cursor-pointer tw-bg-yellow/15 tw-text-yellow tw-gap-10 tw-text-16 tw-font-700 tw-rounded-8 tw-h-[48px] tw-py-16 tw-border-solid tw-border-4 tw-border-yellow"
        >
          Complete Bracket
        </button>
      </div>
      <div
        class="tw-flex tw-items-center tw-justify-center tw-gap-[16px] tw-mt-[52px]"
      >
        <input
          class="tw-appearance-none tw-rounded-8 tw-cursor-pointer tw-border tw-border-solid tw-border-white"
          id="notify-participants-check"
          style="background-image: none; height: 24px; width: 24px;"
          type="checkbox"
        />
        <label
          class="tw-font-500 tw-text-16 tw-items-center"
          for="notify-participants-check"
        >
          Notify Participants
        </label>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
